x-pack/plugins/secops/public/components/notes/add_note/index.tsx

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiButton, EuiButtonEmpty, EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import * as React from 'react';
+import { pure } from 'recompose';
+import styled from 'styled-components';
+
+import {
+  AssociateNote,
+  GetNewNoteId,
+  updateAndAssociateNode,
+  UpdateInternalNewNote,
+  UpdateNote,
+} from '../helpers';
+import * as i18n from '../translations';
+import { NewNote } from './new_note';
+
+const AddNotesContainer = styled(EuiFlexGroup)`
+  margin-bottom: 5px;
+  user-select: none;
+`;
+
+const ButtonsContainer = styled(EuiFlexGroup)`
+  margin-top: 5px;
+`;
+
+export const CancelButton = pure<{ onCancelAddNote: () => void }>(({ onCancelAddNote }) => (
+  <EuiButtonEmpty data-test-subj="cancel" onClick={onCancelAddNote}>
+    {i18n.CANCEL}
+  </EuiButtonEmpty>
+));
+
+/** Displays an input for entering a new note, with an adjacent "Add" button */
+export const AddNote = pure<{
+  associateNote: AssociateNote;
+  getNewNoteId: GetNewNoteId;
+  newNote: string;
+  onCancelAddNote?: () => void;
+  updateNewNote: UpdateInternalNewNote;
+  updateNote: UpdateNote;
+}>(({ associateNote, getNewNoteId, newNote, onCancelAddNote, updateNewNote, updateNote }) => (
+  <AddNotesContainer alignItems="flexEnd" direction="column" gutterSize="none">
+    <NewNote note={newNote} noteInputHeight={200} updateNewNote={updateNewNote} />
+    <ButtonsContainer gutterSize="none">
+      {onCancelAddNote != null ? (
+        <EuiFlexItem>
+          <CancelButton onCancelAddNote={onCancelAddNote} />
+        </EuiFlexItem>
+      ) : null}
+      <EuiFlexItem>
+        <EuiButton
+          data-test-subj="add-note"
+          isDisabled={newNote.trim().length < 1}
```

Comment:
Optional, this reads funny to me. isn't this `=== 0`? But since we use `.length > 0` everywhere it's not that big of deal. Just reads a  bit funny.